api/api/models/category.py

```diff
@@ -0,0 +1,15 @@
+from mongoengine import Document, ReferenceField, ListField, StringField
+from hardware import Hardware #noqa
+
+
+class Category(Document):
+    name = StringField(max_length=120)
+    products = ListField(ReferenceField('Hardware'))
+
+
+def get_all_categories():
+    return Category.objects.all()
+
+
+def get_category_by_id(id_):
```

Comment:
Well it's either `http://localhost/category/motherboard/products/{productId}` or `http://localhost/category/{motherboardCategoryId}/products/{productId}` where I think using the name instead of the id makes more sense and is better readable for humans. I don't think we want duplicate category names anyway.

We're using it when we're requesting the category itself, every product in the category or a specific product in the category.